idx/memory/partitioned_idx.go

```diff
@@ -304,6 +304,38 @@ func (p *PartitionedMemoryIdx) FindByTag(orgId uint32, query tagquery.Query) []i
 	return response
 }
 
+func (p *PartitionedMemoryIdx) FindTerms(orgID uint32, tags []string, query tagquery.Query) (uint32, map[string]map[string]uint32) {
+	g, _ := errgroup.WithContext(context.Background())
+	var total uint32
+	results := make([]map[string]map[string]uint32, len(p.Partition))
+	var i int
+	for _, m := range p.Partition {
+		pos, m := i, m
+		g.Go(func() error {
+			numSeries, terms := m.FindTerms(orgID, tags, query)
+			atomic.AddUint32(&total, numSeries)
+			results[pos] = terms
```

Comment:
don't we have to pass `pos` as a function arg.

Reply:
I believe it's bound to the closure. I also think this is why the copy is made. This code is following the pattern in the above `FindByTag` function.

Reply:
you're right.
see https://play.golang.org/p/_HQ3zQZ6Fpg